src/easylink/implementation.py

```diff
@@ -1,27 +1,17 @@
 from pathlib import Path
-from typing import TYPE_CHECKING, Dict, List, Optional
-
-if TYPE_CHECKING:
-    from easylink.configuration import Config
-    from easylink.step import Step
+from typing import Dict, List, Optional
 
 from easylink.utilities import paths
 from easylink.utilities.data_utils import load_yaml
 
 
 class Implementation:
-    def __init__(
-        self,
-        config: "Config",
-        step: "Step",
-    ):
-        self.name = config.get_implementation_name(step.name)
-        self.environment_variables = config.pipeline[step.name]["implementation"][
-            "configuration"
-        ].to_dict()
+    def __init__(self, name: str, step_name: str, environment_variables: Dict[str, str]):
```

Comment:
I just thought it made sense here to *not* pass in config and step directly, as we've pared back what we need from them (rightfully so imo)